File: clients/python/uiform/types/automations/endpoints.py
import nanoid  # type: ignore
from pydantic import BaseModel, Field, computed_field

from ..logs import AutomationConfig, UpdateAutomationRequest
from ..pagination import ListMetadata


class Endpoint(AutomationConfig):
    @computed_field
    @property
    def object(self) -> str:
        return "automation.endpoint"

    id: str = Field(default_factory=lambda: "endp_" + nanoid.generate(), description="Unique identifier for the extraction endpoint")


class ListEndpoints(BaseModel):
    data: list[Endpoint]
    list_metadata: ListMetadata


# Inherits from the methods of UpdateAutomationRequest
class UpdateEndpointRequest(UpdateAutomationRequest):
    pass

File: clients/node/src/types/automations/endpoints.ts
import { AutomationConfig, UpdateAutomationRequest } from '../logs.js';
import { ListMetadata } from '../pagination.js';

export interface Endpoint extends AutomationConfig {
  object: 'automation.endpoint';
  id: string;
}

export interface ListEndpoints {
  data: Endpoint[];
  list_metadata: ListMetadata;
}

export interface UpdateEndpointRequest extends UpdateAutomationRequest {
  // Inherits all properties from UpdateAutomationRequest
}